lib/v2/screens/settings/set_currency/interactor/viewmodels/set_currency_bloc.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:bloc/bloc.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/screens/settings/set_currency/interactor/mappers/rates_state_mapper.dart';
+import 'package:seeds/v2/screens/settings/set_currency/interactor/usecases/get_fiat_rates_use_case.dart';
+import 'package:seeds/v2/screens/settings/set_currency/interactor/viewmodels/bloc.dart';
+
+/// --- BLOC
+class SetCurrencyBloc extends Bloc<SetCurrencyEvent, SetCurrencyState> {
+  SetCurrencyBloc() : super(SetCurrencyState.initial());
+
+  @override
+  Stream<SetCurrencyState> mapEventToState(SetCurrencyEvent event) async* {
+    if (event is OnQueryChanged) {
+      if (event.query.isNotEmpty && event.query.length == 3 && event.query != state.currentQuery) {
```

Comment:
i dont like that this logic is on the `mapEventToState` method. This should be moved to a place where it can be easily tested.

Reply:
Any suggestion? the only thing comes to mind to create an IsValidCall function

Reply:
yeah, lets do that for now